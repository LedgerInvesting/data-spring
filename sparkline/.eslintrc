{
  "parser": "babel-eslint",
  "rules": {
    "func-names": [
      "error",
      "never"
    ],
    "one-var": "off",
  },
  "env": {
    "browser": true
  }
}
