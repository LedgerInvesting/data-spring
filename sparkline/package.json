{
  "name": "@data-spring/sparkline",
  "version": "0.9.11",
  "main": "build/index.js",
  "scripts": {
    "build": "webpack --production",
    "test": "jest"
  },
  "description": "Components and utility functions for using react-spring animations in data-ui/sparkline data series visualizations https://mlcrazy.github.io/data-spring/",
  "repository": "https://github.com/mlcrazy/data-spring.git",
  "keywords": [
    "d3",
    "react",
    "chart",
    "visualization",
    "plot",
    "motion",
    "animated",
    "animation",
    "spring",
    "data-ui",
    "react-spring"
  ],
  "author": "Matt Liu <matt@ledgerinvesting.com>",
  "license": "MIT",
  "devDependencies": {
    "@babel/core": "^7.0.1",
    "@babel/preset-env": "^7.0.0",
    "@babel/preset-react": "^7.0.0",
    "@babel/runtime": "^7.3.1",
    "@data-ui/sparkline": "^0.0.77",
    "babel-eslint": "^8.2.2",
    "babel-loader": "^8.0.2",
    "eslint": "^4.19.1",
    "eslint-config-airbnb": "^16.1.0",
    "eslint-plugin-import": "^2.7.0",
    "eslint-plugin-react": "^7.7.0",
    "jest": "^24.1.0",
    "react": "^16.0.0",
    "react-dom": "^16.0.0",
    "react-spring": "^7.2.8",
    "react-test-renderer": "^16.8.1",
    "webpack": "^4.20.2",
    "webpack-cli": "^3.1.1",
    "webpack-dev-server": "^3.1.8"
  },
  "jest": {
    "testURL": "http://localhost/"
  }
}
