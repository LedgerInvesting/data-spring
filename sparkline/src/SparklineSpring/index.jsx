import React from 'react';
import { Sparkline } from '@data-ui/sparkline';
import { Spring, config } from 'react-spring';
import { calcFlattenedStart, constructArray } from '../utils/motionUtil';


const SparklineSpring = (props) => {
  const motionPoints = calcFlattenedStart(props);

  return (
    <Spring
      {...props}
      config={props.springConfig}
      from={motionPoints.from}
      to={motionPoints.to}
    >
      {values => {
        return (
          <Sparkline
            {...props}
            data={constructArray(values, props.data)}
          >
            {props.children}
          </Sparkline>
        );
      }}
    </Spring>
  );
};

SparklineSpring.displayName = 'SparklineSpring';
SparklineSpring.defaultProps = {
  animationStart: 'bottom',
  springConfig: config.default,
  customAnimationStart: (point) => point,
};
export default SparklineSpring;
