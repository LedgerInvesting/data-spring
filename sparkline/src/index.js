import SparklineSpring from './SparklineSpring/index.jsx';
export {
  SparklineSpring as SparklineSpring,
};
