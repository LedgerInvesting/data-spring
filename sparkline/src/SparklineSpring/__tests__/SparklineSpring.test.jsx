import React from 'react';
import SparklineSpring from '../';
import { config } from 'react-spring';
import { LineSeries, PointSeries } from '@data-ui/sparkline';
import renderer from 'react-test-renderer';


it('renders successfully', () => {
  const tree = renderer
    .create(
      <SparklineSpring
        width={700}
        height={350}
        ariaLabel="Test animated line"
        animationStart="bottom"
        data={testSeries.xySeries}
        bounds={chartBounds}
        springConfig={config.gentle}
      >
        <LineSeries
          strokeWidth={0.3}
          curve="monotoneX"
          fill="#000"
          stroke="#000"
        />
      </SparklineSpring>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a point series', () => {
  const tree = renderer
    .create(
      <SparklineSpring
        width={700}
        height={350}
        ariaLabel="Test animated line"
        animationStart="bottom"
        data={testSeries.xySeries}
        bounds={chartBounds}
        springConfig={config.gentle}
      >
        <PointSeries
          size={5}
          fill="#000"
          stroke="#000"
        />
      </SparklineSpring>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with default animationStart props', () => {
  const tree = renderer
    .create(
      <SparklineSpring
        width={700}
        height={350}
        ariaLabel="Test animated line"
        data={testSeries.xySeries}
        bounds={chartBounds}
        springConfig={config.gentle}
      >
        <LineSeries
          strokeWidth={0.3}
          curve="monotoneX"
          fill="#000"
          stroke="#000"
        />
      </SparklineSpring>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with default springConfig props', () => {
  const tree = renderer
    .create(
      <SparklineSpring
        width={700}
        height={350}
        ariaLabel="Test animated line"
        animationStart="bottom"
        data={testSeries.xySeries}
        bounds={chartBounds}
      >
        <LineSeries
          strokeWidth={0.3}
          curve="monotoneX"
          fill="#000"
          stroke="#000"
        />
      </SparklineSpring>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

const testSeries = {
  xySeries: [
    {
      x: 1,
      y: 9,
    },
    {
      x: 5,
      y: 4,
    },
    {
      x: 8,
      y: 3,
    },
    {
      x: 10,
      y: 2,
    },
  ],
}

const chartBounds = {
  xMin: 0,
  yMin: 0,
  xMax: 9,
  yMax: 10,
};
