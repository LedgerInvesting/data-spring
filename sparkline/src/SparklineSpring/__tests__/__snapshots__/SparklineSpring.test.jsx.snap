// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a point series 1`] = `
<svg
  aria-label="Test animated line"
  className={null}
  height={350}
  preserveAspectRatio={null}
  role="img"
  style={null}
  viewBox={null}
  width={700}
>
  <g
    className="vx-group"
    transform="translate(16, 16)"
  >
    <g
      className="vx-group"
      transform="translate(0, 0)"
    >
      <g
        className="vx-group vx-glyph"
        transform="translate(0, 0)"
      >
        <circle
          className="vx-glyph-dot"
          cx={0}
          cy={159}
          fill="#000"
          fillOpacity={1}
          onMouseLeave={null}
          onMouseMove={null}
          r={5}
          stroke="#000"
          strokeWidth={2}
        />
      </g>
      <g
        className="vx-group vx-glyph"
        transform="translate(0, 0)"
      >
        <circle
          className="vx-glyph-dot"
          cx={222.66666666666666}
          cy={159}
          fill="#000"
          fillOpacity={1}
          onMouseLeave={null}
          onMouseMove={null}
          r={5}
          stroke="#000"
          strokeWidth={2}
        />
      </g>
      <g
        className="vx-group vx-glyph"
        transform="translate(0, 0)"
      >
        <circle
          className="vx-glyph-dot"
          cx={445.3333333333333}
          cy={159}
          fill="#000"
          fillOpacity={1}
          onMouseLeave={null}
          onMouseMove={null}
          r={5}
          stroke="#000"
          strokeWidth={2}
        />
      </g>
      <g
        className="vx-group vx-glyph"
        transform="translate(0, 0)"
      >
        <circle
          className="vx-glyph-dot"
          cx={668}
          cy={159}
          fill="#000"
          fillOpacity={1}
          onMouseLeave={null}
          onMouseMove={null}
          r={5}
          stroke="#000"
          strokeWidth={2}
        />
      </g>
    </g>
  </g>
</svg>
`;

exports[`renders successfully 1`] = `
<svg
  aria-label="Test animated line"
  className={null}
  height={350}
  preserveAspectRatio={null}
  role="img"
  style={null}
  viewBox={null}
  width={700}
>
  <g
    className="vx-group"
    transform="translate(16, 16)"
  >
    <g
      className="vx-group"
      onMouseLeave={null}
      onMouseMove={null}
      transform="translate(0, 0)"
    >
      <g>
        <path
          className="vx-linepath"
          d="M0,159C74.22222222222221,159,148.44444444444446,159,222.66666666666666,159C296.88888888888886,159,371.1111111111111,159,445.3333333333333,159C519.5555555555555,159,593.7777777777778,159,668,159"
          fill="none"
          stroke="#000"
          strokeDasharray={null}
          strokeDashoffset={0}
          strokeLinecap="round"
          strokeWidth={0.3}
        />
      </g>
    </g>
  </g>
</svg>
`;

exports[`renders with default animationStart props 1`] = `
<svg
  aria-label="Test animated line"
  className={null}
  height={350}
  preserveAspectRatio={null}
  role="img"
  style={null}
  viewBox={null}
  width={700}
>
  <g
    className="vx-group"
    transform="translate(16, 16)"
  >
    <g
      className="vx-group"
      onMouseLeave={null}
      onMouseMove={null}
      transform="translate(0, 0)"
    >
      <g>
        <path
          className="vx-linepath"
          d="M0,159C74.22222222222221,159,148.44444444444446,159,222.66666666666666,159C296.88888888888886,159,371.1111111111111,159,445.3333333333333,159C519.5555555555555,159,593.7777777777778,159,668,159"
          fill="none"
          stroke="#000"
          strokeDasharray={null}
          strokeDashoffset={0}
          strokeLinecap="round"
          strokeWidth={0.3}
        />
      </g>
    </g>
  </g>
</svg>
`;

exports[`renders with default springConfig props 1`] = `
<svg
  aria-label="Test animated line"
  className={null}
  height={350}
  preserveAspectRatio={null}
  role="img"
  style={null}
  viewBox={null}
  width={700}
>
  <g
    className="vx-group"
    transform="translate(16, 16)"
  >
    <g
      className="vx-group"
      onMouseLeave={null}
      onMouseMove={null}
      transform="translate(0, 0)"
    >
      <g>
        <path
          className="vx-linepath"
          d="M0,159C74.22222222222221,159,148.44444444444446,159,222.66666666666666,159C296.88888888888886,159,371.1111111111111,159,445.3333333333333,159C519.5555555555555,159,593.7777777777778,159,668,159"
          fill="none"
          stroke="#000"
          strokeDasharray={null}
          strokeDashoffset={0}
          strokeLinecap="round"
          strokeWidth={0.3}
        />
      </g>
    </g>
  </g>
</svg>
`;
